Mathlib/Geometry/Euclidean/Sphere/Basic.lean

```diff
@@ -201,6 +209,120 @@ theorem concyclic_singleton (p : P) : Concyclic ({p} : Set P) :=
 theorem concyclic_pair (p₁ p₂ : P) : Concyclic ({p₁, p₂} : Set P) :=
   ⟨cospherical_pair p₁ p₂, coplanar_pair ℝ p₁ p₂⟩
 
+namespace Sphere
+
+/-- `s.IsDiameter p₁ p₂` says that `p₁` and `p₂` are the two endpoints of a diameter of `s`. -/
+structure IsDiameter (s : Sphere P) (p₁ p₂ : P) : Prop where
+  left_mem : p₁ ∈ s
+  midpoint_eq_center : midpoint ℝ p₁ p₂ = s.center
+
+variable {s : Sphere P} {p₁ p₂ p₃ : P}
+
+lemma IsDiameter.right_mem (h : s.IsDiameter p₁ p₂) : p₂ ∈ s := by
+  rw [mem_sphere, ← mem_sphere.1 h.left_mem, ← h.midpoint_eq_center,
+    dist_left_midpoint_eq_dist_right_midpoint]
+
+protected lemma IsDiameter.symm (h : s.IsDiameter p₁ p₂) : s.IsDiameter p₂ p₁ :=
+  ⟨h.right_mem, midpoint_comm (R := ℝ) p₁ p₂ ▸ h.midpoint_eq_center⟩
+
+lemma isDiameter_comm : s.IsDiameter p₁ p₂ ↔ s.IsDiameter p₂ p₁ :=
+  ⟨IsDiameter.symm, IsDiameter.symm⟩
+
+lemma isDiameter_iff_left_mem_and_midpoint_eq_center :
+    s.IsDiameter p₁ p₂ ↔ p₁ ∈ s ∧ midpoint ℝ p₁ p₂ = s.center :=
+  ⟨fun h ↦ ⟨h.1, h.2⟩, fun h ↦ ⟨h.1, h.2⟩⟩
+
+lemma isDiameter_iff_right_mem_and_midpoint_eq_center :
+    s.IsDiameter p₁ p₂ ↔ p₂ ∈ s ∧ midpoint ℝ p₁ p₂ = s.center :=
+  ⟨fun h ↦ ⟨h.right_mem, h.2⟩, fun h ↦ IsDiameter.symm ⟨h.1, midpoint_comm (R := ℝ) p₁ p₂ ▸ h.2⟩⟩
+
+lemma IsDiameter.pointReflection_center_left (h : s.IsDiameter p₁ p₂) :
+    Equiv.pointReflection s.center p₁ = p₂ := by
+  rw [← h.midpoint_eq_center, Equiv.pointReflection_midpoint_left]
+
+lemma IsDiameter.affine_pointReflection_center_left (h : s.IsDiameter p₁ p₂) :
+    AffineEquiv.pointReflection ℝ s.center p₁ = p₂ := by
```

Comment:
I think I'd argue that the simp-normal form of `AffineEquiv.pointReflection K X Y` should be `Equiv.pointReflection X Y`, and similarly for `.symm`. This is generally how we handled incremental bundling elsewhere - pick the weakest bundling (or a plain function) as the simp-normal form, and aggressively unbundle everything with `simp`.

Reply:
I've made that change to simp-normal form (including removing `simp` markings from lemmas `simp` can now prove) and removed the copies of lemmas about diameters that used `AffineEquiv.pointReflection`.